### DISCUSSION TITLE
Add authentication header after login

### DISCUSSION THREAD

**Discussion by mhmmdsmdi** [canton7/RestEase#197] [Q&A] [answered]
I want to get authentication token by calling IAuthentication.Login(username,password)
and then use the result (Brearer Token) in other interfaces like user

I use this code for my DI

`            services.AddHttpClient("ScadaApiClient")
                .ConfigureHttpClient(x => x.BaseAddress = baseAddress)
                .UseWithRestEaseClient<IAuthenticationApi>()
                .UseWithRestEaseClient<IUserApi>();`

**Comment by canton7** [comment, marked as answer]
You've basically got two options:

1. Add the Authentication header as a property on `IUserApi`. In this case, you'll need to assign that property for every instance of `IUserApi` which is created. There's nothing built-in to let you do this, so you'll have to register `IUserApi` with the IoC container yourself (without using `UseWithRestEaseClient`), and add the necessary code to assign to that Authentication header.
2. Add the Authentication header using a custom `HttpMessageHandler`. You can then register that, using e.g. `.AddHttpClient("...").ConfigurePrimaryHttpMessageHandler(...).UseWithRestEaseClient<IUserApi>()`.

**Reply by mhmmdsmdi** [reply]
tanks for reply.

can you give me a sample to understand what should going to do exactly
